ProcessList.c

```diff
@@ -60,6 +63,7 @@ void ProcessList_done(ProcessList* this) {
    Hashtable_delete(this->processTable);
    Vector_delete(this->processes);
    Vector_delete(this->processes2);
+   Hashtable_delete(this->treeSet);
```

Comment:
Delete weak pointers (the treeSet) before the hard (owning) links (actual process list). That way you avoid a race condition where there are pointers around for memory that has been freed.